src/routes/tools/local-climate-change-snapshot/_helpers.js

```diff
@@ -1,17 +1,53 @@
 import { groups, merge, mean, rollups, extent } from "d3-array";
 
 // Helpers
-import { buildEnvelope } from "../_common/helpers";
+import { buildEnvelope, setInitialLocation } from "../_common/helpers";
 
 // Constants
-import { DEFAULT_STAT_PERIODS } from "../_common/constants";
+import { DEFAULT_STAT_PERIODS, DEFAULT_LOCATION } from "../_common/constants";
 import {
   ENVELOPE_SEARCH_EXP,
   AVERAGE_SEARCH_EXP,
   SCENARIOS,
   SCENARIO_RANGES,
+  VALID_BOUNDARY_TYPES,
 } from "./_constants";
 
+export const isValidNumber = (value) =>
+  typeof value === "number" && !isNaN(value);
+
+export const isValidLocationParams = (lng, lat, boundaryType) => {
+  return (
+    isValidNumber(lat) &&
+    isValidNumber(lng) &&
+    VALID_BOUNDARY_TYPES.has(boundaryType)
+  );
+};
+
+export const getLocationFromQuery = async (
+  lng,
+  lat,
+  boundaryType,
+  useDefaultLocation = false
```

Comment:
could make this default param set to `DEFAULT_LOCATION` and then make the function caller deliberately set it to `false` which would eliminate the need for the `fallbackLocation` variable within the function.